packages/@lwc/errors/src/compiler/error-info/template-transform.ts

```diff
@@ -464,17 +464,26 @@ export const ParserDiagnostics = {
         level: DiagnosticLevel.Warning,
         url: '',
     },
-    ATTRIBUTE_CANNOT_START_OR_END_WITH_UNDERSCORE: {
+
+    ATTRIBUTE_NAME_MUST_START_WITH_ALPHABETIC_CHARACTER: {
         code: 1124,
         message:
-            '{0} is not valid attribute for {1}. Attribute name cannot start or end with an underscore.',
+            '{0} is not valid attribute for {1}. Attribute name cannot start with non alphabetic character.',
```

Comment:
probably avoid negation on this messag,e something like: `Attribute name must start with alphabetic character.` instead